File: src/main/java/org/voegtle/weatherwidget/location/LocationIdentifier.java
package org.voegtle.weatherwidget.location;

public enum LocationIdentifier {
  Paderborn("tegelweg8"), BadLippspringe("bali"), Leopoldshoehe("leoxity"), Bonn("forstweg17"), Magdeburg("elb"), Herzogenaurach("herzo"), Freiburg("ochsengasse"), Mobil("mobil"), Shenzhen("SZ");
  private final String id;

  LocationIdentifier(String id) {
    this.id = id;
  }

  public static LocationIdentifier getByString(String stringIdentifier) {
    for (LocationIdentifier identifier : values()) {
      if (identifier.id.equals(stringIdentifier)) {
        return identifier;
      }
    }
    return null;
  }
}

File: src/main/java/org/voegtle/weatherwidget/location/LocationIdentifier.kt
package org.voegtle.weatherwidget.location

enum class LocationIdentifier(private val id: String) {
  Paderborn("tegelweg8"), Paderborn20("wetterwolke"), BadLippspringe("bali"), Leopoldshoehe("leoxity"), Bonn("forstweg17"), Magdeburg("elb"), Herzogenaurach("herzo"), Freiburg("ochsengasse"), Mobil("mobil"), Shenzhen("SZ");

  companion object {
    fun getByString(stringIdentifier: String): LocationIdentifier? {
      return values().firstOrNull { it.id == stringIdentifier }
    }
  }
}
